lib/libvfio-user.c

```diff
@@ -1288,10 +1295,16 @@ vfu_create_ctx(vfu_trans_t trans, const char *path, int flags, void *pvt)
         return NULL;
     }
 
+    if (dev_type != VFU_DEV_TYPE_PCI) {
+        errno = EINVAL;
```

Comment:
Should this be ENOTSUP?

Reply:
No, `ENOTSUP` should be for thing we _could_ support but we don't. The user might set `dev_type` to something completely bogus.

Reply:
But today we only support PCI and isn't the idea of adding device_type is to say we will support non-pci in future.

Reply:
When we add another device type we'll document it, so if the user supplies a wrong value then it's an invalid argument.